File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int ans = nums[0];
        int currentSum = 0;
        for(int i = 0; i<n; i++){
            
            currentSum += nums[i];
            ans = max(ans, currentSum);
            if(currentSum < 0) currentSum = 0;
        }
        return ans;
    }
};

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currSum = 0
        maxSum = -sys.maxsize-1
        n = len(nums)
        for i in range(0, n):
            currSum += nums[i]
            maxSum = max(currSum, maxSum)
            if currSum < 0:
                currSum = 0
        return maxSum